rfc/rfc-80/rfc-80.md

```diff
@@ -0,0 +1,161 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+# RFC-80: Support column families for wide tbles
+
+## Proposers
+
+- @xiarixiaoyao
+- @wombatu-kun
+
+## Approvers
+ - 
+ - 
+
+## Status
+
+JIRA: https://issues.apache.org/jira/browse/HUDI-
+
+## Abstract
+
+In streaming processing, there are often scenarios where the table is widened. The current mainstream real-time stretching is completed through Flink's multi-layer join;
+Flink's join will cache a large amount of data in the state backend. As the data set increases, the pressure on the Flink task state backend will gradually increase, and may even become unavailable.
```

Comment:
Since 1.x release, we support NB-CC mode that supports multiple jobs ingesting to the same table concurrently, so that we can get rid of the join and the payloads would got filled up during compaction, does that ease the scenario here?

Reply:
As this overall design relies on NB-CC support, it is necessary anyway. I don't think that it can ease a lot for the scenario when multiple writers ingest their own subset of columns while readers want to read different subsets or the whole set of columns. Both writing, reading and compacting will be complicated and not very performant. And if we separate ingestion to different tables, we'll have to join them on reading.

Reply:
> for the scenario when multiple writers ingest their own subset of columns while readers want to read different subsets or the whole set of columns.

So the main gains here is for reader columns pruning?

Reply:
> > for the scenario when multiple writers ingest their own subset of columns while readers want to read different subsets or the whole set of columns.
> 
> So the main gains here is for reader columns pruning?

Main gains of clustering columns for wide tables are:  

**Write performance**:  
1) Writing is similar to ordinary bucket writing, but it involves splitting column clusters and sorting. Therefore, the writing performance of full data is 10% lower than that of native bucket writing.  
2) However, if only some columns are updated among a large number of columns, the writing efficiency is much faster than that of non-column clustered tables.  

**Read performance**:  
Since the data is already sorted when it is written, the SortMerge method can be used directly to merge the data; compared with the native bucket data reading performance is improved a lot, and the memory consumption is reduced significantly.  

**Compaction performance**:  
The logic of compaction and reading is the same.

Reply:
Sorry, I didn't get the "write performance" part, for mor table with bucket index, the writer only flushes the records into avros which is already fast, how could the column cluster and sort be faster? As for partial update, the payloads would got merged either way on reader side(including compaction).

Reply:
ok, added this comment to RFC itself.

Reply:
> 1.x also supports partial updates being encoded in logfiles. So that should be able to handle this scenario already?  

Yes, partial update is a kinda alternative approach for handling this scenario.

Reply:
> we need to introduce new log headers or standard mechanism for merging to determine if base file or log files are sorted

We will introduce a merge sort for log compaction in RFC-81, should be very related.